plugins/colorsource/export.js

```diff
@@ -0,0 +1,332 @@
+    "B28E1514-AE8C-4E06-8472-B52D575B1CF2": [`Red`, `Green`, `Blue`, `White`, `UV`],
+    "77597794-7BFF-46A3-878B-906D3780E6C9": [`Red`, `Blue`, `White`, `Indigo`],
+    "77A82F8A-9B24-4C3F-98FC-B6A29FB1AAE6": [`Red`, `Green`, `Blue`, `White`],
+    "74EF89F4-0B78-4DC6-8E8A-68E3298B7CD2": [`Red`, `Green`, `Blue`, `UV`],
+    "D3E71EC8-3406-4572-A64C-52A38649C795": [`Red`, `Green`, `Blue`, `Amber`],
+    "B043D095-95A4-4DDB-AB38-252C991B13A8": [`Red`, `Green`, `Blue`, `Indigo`],
+    "3874B444-A11E-47D9-8295-04556EAEBEA7": [`Red`, `Green`, `Blue`],
+    "637E8789-5540-45D5-BD83-D7C2A7618B45": [`Red`, `Green`, `Indigo`],
+    "EF4970BA-2536-4725-9B0F-B2D7A021E139": [`Cyan`, `Magenta`, `Yellow`],
+    "E6AC63D6-1349-4BFC-9A04-7548D1DB8E1F": [`CoolWhite`, `MediumWhite`, `WarmWhite`],
+    "7B365530-A4DF-44AD-AEF5-225472BE02AE": [`CoolWhite`, `WarmWhite`],
+    "B074A2D3-0C40-45A7-844A-7C2721E0B267": [`Hue`, `Saturation`]
+  };
+
+  let selectedColorTable = Object.keys(colorTables).find(
+    colorTable => colorTables[colorTable].every(
+      color => colorChannels.some(ch => ch.name === color)
+    )
+  );
+
+  const has16bitHue = colorChannels.some(ch => ch.name === `Hue` && ch.size === 16);
+  if (selectedColorTable === `B074A2D3-0C40-45A7-844A-7C2721E0B267` && has16bitHue) {
+    selectedColorTable = `B3D05F0E-FB45-4EEA-A8D5-61F545A922DE`; // this is a special case; it refers to Hue / Hue fine / Saturation
+  }
+
+  return selectedColorTable || null;
+}
+
+/**
+ * @param {Mode} mode The personality's mode.
+ * @returns {Array.<Object>} Array of ColorSource commands (like "Reset"), may be empty. The commands are generated by Maintenance capabilities with hold time set.
+ */
+function getCommands(mode) {
+  const commands = [];
+  mode.channels.forEach((channel, channelIndex) => {
+    if (!channel.capabilities) {
+      // e. g. fine channels
+      return;
+    }
+
+    channel.capabilities.forEach(cap => {
+      if (cap.type === `Maintenance` && cap.hold) {
+        commands.push({
+          name: cap.comment,
+          steps: [
+            {
+              actions: [{
+                dmx: channelIndex,
+                value: cap.getMenuClickDmxValueWithResolution(CoarseChannel.RESOLUTION_8BIT)
+              }],
+              wait: 0 // this is apparently the delay before this step is activated
+            },
+            {
+              actions: [{
+                dmx: channelIndex,
+                value: -1
+              }],
+              wait: cap.hold.getBaseUnitEntity().number
+            }
+          ]
+        });
+      }
+    });
+  });
+
+  return commands;
+}
+
+/**
+ * @param {Mode} mode The personality's mode.
+ * @param {Boolean} hasIntensity Whether the mode has an intensity channel. Should equal fixtureJson.hasIntensity
+ * @returns {Array.<Object>} ColorSource channel objects of all mode channels.
+ */
+function getCSChannels(mode, hasIntensity) {
```

Comment:
Please rename this to either `getCsChannels` (similar to how we would have `getOflChannels`) or `getColorSourceChannels`. In the latter case, we should probably include this in all function names.